dissect/target/plugins/general/plugins.py

```diff
@@ -59,16 +59,26 @@ def generate_functions_json(functions: list[plugin.FunctionDescriptor] | None =
 
     for desc in functions or _get_default_functions():
         docstring = desc.func.__doc__.split("\n\n", 1)[0].strip() if desc.func.__doc__ else None
-        arguments = [
-            {
+        arguments = []
+
+        for name, _arg in desc.args:
+            is_bool_action = _arg.get("action", "") in ("store_true", "store_false")
+            arg_desc = {
                 "name": name[0],
-                "type": getattr(arg.get("type"), "__name__", None),
-                "help": arg.get("help"),
-                "default": arg.get("default"),
-                "required": arg.get("required", False),
+                # infer the type either by store_*, type argument and fallback to default str.
+                # See: https://docs.python.org/3/library/argparse.html#type
+                "type": "bool" if is_bool_action else getattr(_arg.get("type"), "__name__", "str"),
+                "help": _arg.get("help"),
+                "default": _arg.get("default") or _arg.get("const"),
+                # required can either be set explicitly or is implied with '--' style arguments.
+                "required": (name[0].startswith("--") and not _arg.get("default") and not is_bool_action)
+                or _arg.get("required", False),
```

Comment:
```suggestion
                "required": _arg.get("required", False),
```
Only `--` style arguments are currently allowed anyway? And since `False` will always be the default because of this, this can only ever be `True` if explicitly specified, right?